### PR TITLE
refactor multi_choice_to_labels

### LINKED ISSUE #48: [BUG] Improve REDCapTidieR Execution Times and Benchmarking
# Expected Behavior
`REDCapTidieR` should run relatively quickly (<10s) on most manageable databases. 

# Current Behavior
Currently `REDCapTidieR` can take long stretches of time that are not related to API calls. Using `profvis` it was found that the main culprit here is the `check_repeat_and_nonrepeat` function that gets looped through inside of the `clean_*` functions. This is a nested `for` loop that looks across rows and columns of a database to check whether a project is repeating or nonrepeating.

The complicating factor here is the allowed REDCap behavior to design an instrument that can be _both_ repeating **and** nonrepeating. This is a behavior we do not want to support, but wish to loudly fail on. If we can fix the below chunk for this check, it should greatly improve execution time:

https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/1c70017dff24b1600bfe18d79695af3ff148ae5e/R/checks.R#L67-L89

There may be some other areas that can marginally improve execution time, but this seems to be the biggest driving factor.

**Screenshots**
`profvis` outputs shown below:
image
image

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR addresses the final bottleneck with `multi_choice_to_labels()` discussed in https://github.com/CHOP-CGTDataOps/REDCapTidieR/issues/48#issuecomment-1285717234. The savings come from:
1. using `dplyr::recode()` rather than a combo of `left_join() %>% mutate() %>% select()`
2. updating columns in the data directly  rather than with `mutate()`:
```r
# This
data[[my_field]] <- f(data[[my_field]])

# Not this
data <- mutate(data, my_field = f(my_field))
```

All together our runtime on prodigy is below 7 seconds:
```
Unit: seconds                                                                                                                                                                       
                         expr      min       lq     mean   median       uq     max neval
 read_redcap_tidy(uri, token) 5.729213 6.512654 6.938895 6.885623 7.231833 8.87483    10
```

and we're spending the bulk of our time on API calls and processing the data (see `profvis` output below)

# Proposed Changes 
List changes below in bullet format:

- Add a `return_vector` parameter to `parse_labels()` to allow it to return an object that can be directly used by `recode()`
    - The default is still `return_vector = FALSE` which returns a `tibble`
- Refactor `multi_choice_to_labels()` to use `dplyr::recode()`

**Screenshots**
Screen Shot 2022-10-27 at 1 28 57 PM

### Issue Addressed
Closes #48 

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [x] I checked that new files follow naming conventions and are in the right place
- [x] I checked that documentation is complete, clear, and without typos
- [x] I added/edited comments to explain "why" not "how"
- [x] I checked that all new variable and function names follow naming conventions
- [x] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [x] I checked that new tests address important edge cases

### CODE REVIEW

**Review by rsh52 — APPROVED** [review]
Love the condensing you did with `recode`, well done. Impressed that the overall `for` loop wasn't the culprit and it was instead the `mutate` statements.

Looks good on my end, there may be some merge conflicts to resolve between the couple PRs here but we can address as we move forward.

File: R/utils.R [hunk 1]
```diff
@@ -80,18 +80,21 @@ link_arms <- function(
...
+#' @param return_vector logical for whether to return result as a vector
 #'
 #' @importFrom stringi stri_split_fixed
 #' @importFrom tibble as_tibble is_tibble
 #' @importFrom rlang .data
 #'
 #' @keywords internal
 
-parse_labels <- function(string) {
+parse_labels <- function(string, return_vector = FALSE) {
```
Comment (skadauke) [inline R/utils.R:97]: Since this is an internal function I think we can safely assume that nobody depends on the old behavior? Do we need the `return = FALSE` code anywhere else? If not, I would scrap it.

Comment (ezraporter) [reply R/utils.R:97]: Yeah we use it here:
https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/2e445b99a036e036e02d2d8b0c3b2343eb645bbd/R/utils.R#L173-L183

which is used to add all the distinct checkbox option fields to the metadata. Could probably refactor that to use the vector output from `parse_labels()` rather than `tibble` but I'd also just as happily not touch it. What do you think?

Comment (skadauke) [reply R/utils.R:97]: I think we can leave as is then!

**Review by skadauke — APPROVED** [review]
